packages/smooth_app/lib/helpers/picture_capture_helper.dart

```
@@ -1,10 +1,106 @@
+          barcode: inputTask.barcode,
+          imageField: ImageFieldExtension.getType(inputTask.imageField),
+          imageUri: Uri.parse(inputTask.imageUri),
+        );
+        final Status result = await OpenFoodAPIClient.addProductImage(
+          ProductQuery.getUser(),
+          image,
+        );
+        shouldRetry = result.error != null || result.status != 'status ok';
+      } catch (e) {
+        shouldRetry = true;
+      }
+      if (shouldRetry) {
+        inputTask.counter += 1;
+        await Workmanager().initialize(callbackDispatcher);
+        await Workmanager().registerOneOffTask(
+          task,
+          'ImageUploadWorker',
+          constraints: Constraints(
+            networkType: NetworkType.connected,
+          ),
+          inputData: inputTask.toJson(),
+          initialDelay: duration[counter - 1],
+        );
+        return Future<bool>.error('Failed and it will try again');
+      } else {
+        // go to the file system and delete the file that was uploaded
+        final File file = File(inputData['imageUri'].toString());
+        file.deleteSync();
+        final LocalDatabase localDatabase =
+            await LocalDatabase.getLocalDatabase();
+        final DaoProduct daoProduct = DaoProduct(localDatabase);
+        final ProductQueryConfiguration configuration =
+            ProductQueryConfiguration(
+          inputTask.barcode,
+          fields: ProductQuery.fields,
+          language: LanguageHelper.fromJson(inputTask.languageCode),
+          country: ProductQuery.getCountry(),
+        );
+        try {
+          final ProductResult result =
+              await OpenFoodAPIClient.getProduct(configuration);
+          if (result.status == 1) {
+            final Product? product = result.product;
+            if (product != null) {
+              await daoProduct.put(product);
+            }
+          }
+        } catch (e) {
+          debugPrint('Error: $e,Updating to localdatabse failed');
```

Maybe throw a error here so that we get to see it on Sentry to have a overview if and when it fails

---

Not sure if the sentry will work with work manager, i will research it then i could do it later